packages/twenty-front/src/modules/object-record/record-board/hooks/useUpsertBoardOpportunityRecord.ts

```diff
@@ -0,0 +1,60 @@
+import { useContext } from 'react';
+import { useRecoilCallback } from 'recoil';
+import { isDefined } from '~/utils/isDefined';
+
+import { useRecordBoardColumnContextOrThrow } from '@/object-record/record-board/record-board-column/contexts/RecordBoardColumnContext';
+import { RecordForSelect } from '@/object-record/relation-picker/types/RecordForSelect';
+import { useRecoilComponentCallbackStateV2 } from '@/ui/utilities/state/component-state/hooks/useRecoilComponentCallbackStateV2';
+import { RecordBoardContext } from '../contexts/RecordBoardContext';
+import { recordBoardPendingRecordIdByColumnComponentFamilyState } from '../states/recordBoardPendingRecordIdByColumnComponentFamilyState';
+
+export const useUpsertBoardOpportunityRecord = (recordBoardId: string) => {
```

Comment:
This looks duplicated with useUpsertBoardRecord below. I'm not fully against it if it's hard to merge them (I would prefer two clear hooks over one which is hard to maintin) but I feel we could try to merge them.

Long term, we plan to address record creation through the RightDrawer so I would not over invest on this area. The abstract reason why for "opportunity" we need to fill the "company" is that "company" would be a REQUIRED field on opportunity. Right now it's hard coded for opportunity and it's fine keeping it like this as long as we haven't introduced both required relations and creation in Right Drawer.

So I would not over engineer this and why not just making company optional in useUpsertBoardRecord for now and merge both hooks